### PR TITLE
add big query log processor

### LINKED ISSUE #31153: log() - logarithm function calculates wrong result in BigQuery because parameters backwards
### Describe the bug

- Using the Metabase `log()` function with BigQuery calculates the wrong result because Metabase mixes up the parameters
- As per [these BigQuery docs](https://cloud.google.com/bigquery/docs/reference/standard-sql/mathematical_functions#log), `log()` should be used like `log(x, base)`, but Metabase is sending `log(base, x)` to BigQuery resulting in incorrect results.

### To Reproduce

1. Create a Question on a BigQuery table
2. Add a custom expression called log10 - `log(10)`
3. Add a custom expression called log100 - `log(100)`
4. Add a custom expression called log1000 - `log(1000)`
5. Click Visualize to see the results of the custom expressions
image

Values:
  - log10 should be 1
  - log100 should be 2
  - log1000 should be 3

6. Click View SQL and see that x and base are reversed:
```
SELECT `zendesk_16700.datesAndAmounts`.`date` AS `date`, `zendesk_16700.datesAndAmounts`.`amount` AS `amount`, log(10, 10) AS `log10`, log(10, 100) AS `log100`, log(10, 1000) AS `log1000` FROM `zendesk_16700.datesAndAmounts`
```

### Expected behavior

- Should use BigQuery `log()` function like `log(x, base)` instead of `log(base, x)`

### Logs

_No response_

### Information about your Metabase installation

```JSON
- Metabase version: v1.46.4
```

### Severity

P1 - producing incorrect result with no error or warning

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by sansolovyov**
Fixes #31153. I took over PR #31246, just made a few changes to make it work.

### CODE REVIEW

**Review by metamben — APPROVED** [review]
LGTM

File: modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj [hunk 1]
```diff
@@ -1077,3 +1077,9 @@
                1234.1234567890124
                1234.1234567890123456M]]
              (mt/rows (mt/process-query query)))))))
+
```
Comment (qnkhuat) [inline modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk/query_processor_test.clj:1080]:
```suggestion
```

**Review by qnkhuat — APPROVED** [review]
nice!

**Comment by paoliniluis** [comment]
any reason why this is not backported?

**Pull request merged**